### ISSUE TITLE
My trained lora did not affect the video. What could be the issue?

### ISSUE THREAD

**Issue by taekim99** [kohya-ss/musubi-tuner#121]
Hello, I trained 2 hunyuan video loras so far and it looks like none of them affect the video. Can you help me what was the problem? during generation loras, it did not give any errors and process was very smooth. Below is the command I gave: 

```
python cache_latents.py 
--dataset_config \Downloads\text-to-video-training\musubi-tuner\dataset.toml 
--vae \Downloads\text-to-image-video\ComfyUI\models\vae\pytorch_model.pt 
--vae_chunk_size 32 
--vae_tiling

python cache_text_encoder_outputs.py 
--dataset_config \Downloads\text-to-video-training\musubi-tuner\dataset.toml  
--text_encoder1 \Downloads\text-to-image-video\ComfyUI\models\text_encoders\llava_llama3_fp16.safetensors 
--text_encoder2 \Downloads\text-to-image-video\ComfyUI\models\text_encoders\clip_l_hunyuan.safetensors 
--batch_size 16

accelerate launch 
--num_cpu_threads_per_process 1 
--mixed_precision bf16 
hv_train_network.py 
--dit \Downloads\text-to-image-video\ComfyUI\models\diffusion_models\mp_rank_00_model_states.pt 
--dataset_config \Downloads\text-to-video-training\musubi-tuner\dataset.toml 
--sdpa 
--mixed_precision bf16 
--fp8_base 
--optimizer_type adamw8bit 
--learning_rate 0.0006 
--gradient_checkpointing 
--max_data_loader_n_workers 2 
--persistent_data_loader_workers 
--network_module networks.lora 
--network_dim 32 
--timestep_sampling shift 
--discrete_flow_shift 3.0 
--max_train_epochs 20 
--save_every_n_epochs 5 
--seed 42 
--output_dir \Downloads\text-to-video-training\trained_lora --output_name chj-lora

python convert_lora.py 
--input \Downloads\text-to-video-training\trained_lora\chj-lora.safetensors 
--output \Downloads\text-to-image-video\ComfyUI\models\loras\hunyuan_trained\chj.safetensors
--target other
```

and this is the actual commands and prints I see in cmd:

```
PS \Downloads\text-to-video-training\musubi-tuner> python cache_latents.py --dataset_config \Downloads\text-to-video-training\musubi-tuner\dataset.toml --vae \Downloads\text-to-image-video\ComfyUI\models\vae\pytorch_model.pt --vae_chunk_size 32 --vae_tiling
INFO:__main__:Load dataset config from \Downloads\text-to-video-training\musubi-tuner\dataset.toml
INFO:dataset.image_video_dataset:glob images in /Downloads/text-to-video-training/training_data/chj
INFO:dataset.image_video_dataset:found 25 images
INFO:dataset.config_utils:[Dataset 0]
  is_image_dataset: True
  resolution: (960, 544)
  batch_size: 1
  num_repeats: 4
  caption_extension: ".txt"
  enable_bucket: True
  bucket_no_upscale: False
  cache_directory: "/Downloads/text-to-video-training/training_data/cha"
  debug_dataset: False
    image_directory: "/Downloads/text-to-video-training/training_data/chj"
    image_jsonl_file: "None"


INFO:hunyuan_model.vae:Loading 3D VAE model (884-16c-hy) from: \Downloads\text-to-image-video\ComfyUI\models\vae\pytorch_model.pt
INFO:hunyuan_model.vae:VAE to dtype: torch.float16
INFO:__main__:Loaded VAE: FrozenDict({'in_channels': 3, 'out_channels': 3, 'down_block_types': ['DownEncoderBlockCausal3D', 'DownEncoderBlockCausal3D', 'DownEncoderBlockCausal3D', 'DownEncoderBlockCausal3D'], 'up_block_types': ['UpDecoderBlockCausal3D', 'UpDecoderBlockCausal3D', 'UpDecoderBlockCausal3D', 'UpDecoderBlockCausal3D'], 'block_out_channels': [128, 256, 512, 512], 'layers_per_block': 2, 'act_fn': 'silu', 'latent_channels': 16, 'norm_num_groups': 32, 'sample_size': 256, 'sample_tsize': 64, 'scaling_factor': 0.476986, 'force_upcast': True, 'spatial_compression_ratio': 8, 'time_compression_ratio': 4, 'mid_block_add_attention': True, '_use_default_values': ['force_upcast', 'spatial_compression_ratio'], '_class_name': 'AutoencoderKLCausal3D', '_diffusers_version': '0.4.2'}), dtype: torch.float16
INFO:__main__:Set chunk_size to 32 for CausalConv3d in VAE
INFO:__main__:Encoding dataset [0]
25it [00:23,  1.05it/s]
```

```
PS \Downloads\text-to-video-training\musubi-tuner> python cache_text_encoder_outputs.py --dataset_config \Downloads\text-to-video-training\musubi-tuner\dataset.toml  --text_encoder1 \Downloads\text-to-image-video\ComfyUI\models\text_encoders\llava_llama3_fp16.safetensors --text_encoder2 \Downloads\text-to-image-video\ComfyUI\models\text_encoders\clip_l_hunyuan.safetensors --batch_size 16
INFO:__main__:Load dataset config from \Downloads\text-to-video-training\musubi-tuner\dataset.toml
INFO:dataset.image_video_dataset:glob images in /Downloads/text-to-video-training/training_data/chj
INFO:dataset.image_video_dataset:found 25 images
INFO:dataset.config_utils:[Dataset 0]
  is_image_dataset: True
  resolution: (960, 544)
  batch_size: 1
  num_repeats: 4
  caption_extension: ".txt"
  enable_bucket: True
  bucket_no_upscale: False
  cache_directory: "/Downloads/text-to-video-training/training_data/cha"
  debug_dataset: False
    image_directory: "/Downloads/text-to-video-training/training_data/chj"
    image_jsonl_file: "None"


INFO:__main__:loading text encoder 1: \Downloads\text-to-image-video\ComfyUI\models\text_encoders\llava_llama3_fp16.safetensors
INFO:hunyuan_model.text_encoder:Loading text encoder model (llm) from: \Downloads\text-to-image-video\ComfyUI\models\text_encoders\llava_llama3_fp16.safetensors
INFO:hunyuan_model.text_encoder:Text encoder to dtype: torch.float16
INFO:hunyuan_model.text_encoder:Loading tokenizer (llm) from: \Downloads\text-to-image-video\ComfyUI\models\text_encoders\llava_llama3_fp16.safetensors
INFO:hunyuan_model.text_encoder:Loading tokenizer from Hugging Face: xtuner/llava-llama-3-8b-v1_1-transformers
You are using the default legacy behaviour of the <class 'transformers.models.llama.tokenization_llama_fast.LlamaTokenizerFast'>. This is expected, and simply means that the `legacy` (previous) behavior will be used so nothing changes for you. If you want to use the new behaviour, set `legacy=False`. This should only be set if you understand what it means, and thoroughly read the reason why this was added as explained in https://github.com/huggingface/transformers/pull/24565 - if you loaded a llama tokenizer from a GGUF file you can ignore this message.
INFO:__main__:Encoding with Text Encoder 1
INFO:__main__:Encoding dataset [0]
25it [00:05,  4.41it/s]
INFO:__main__:loading text encoder 2: \Downloads\text-to-image-video\ComfyUI\models\text_encoders\clip_l_hunyuan.safetensors
INFO:hunyuan_model.text_encoder:Loading text encoder model (clipL) from: \Downloads\text-to-image-video\ComfyUI\models\text_encoders\clip_l_hunyuan.safetensors
INFO:hunyuan_model.text_encoder:Text encoder to dtype: torch.float16
INFO:hunyuan_model.text_encoder:Loading tokenizer (clipL) from: \Downloads\text-to-image-video\ComfyUI\models\text_encoders\clip_l_hunyuan.safetensors
INFO:hunyuan_model.text_encoder:Loading tokenizer from Hugging Face: openai/clip-vit-large-patch14
INFO:__main__:Encoding with Text Encoder 2
INFO:__main__:Encoding dataset [0]
25it [00:00, 35.46it/s]
```

```
PS \Downloads\text-to-video-training\musubi-tuner> accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 hv_train_network.py --dit \Downloads\text-to-image-video\ComfyUI\models\diffusion_models\mp_rank_00_model_states.pt --dataset_config \Downloads\text-to-video-training\musubi-tuner\dataset.toml --sdpa --mixed_precision bf16 --fp8_base --optimizer_type adamw8bit --learning_rate 0.0006 --gradient_checkpointing --max_data_loader_n_workers 2 --persistent_data_loader_workers --network_module networks.lora --network_dim 32 --timestep_sampling shift --discrete_flow_shift 3.0 --max_train_epochs 20 --save_every_n_epochs 5 --seed 42 --output_dir \Downloads\text-to-video-training\trained_lora --output_name chj-lora
The following values were not passed to `accelerate launch` and had defaults used instead:
        `--num_processes` was set to a value of `1`
        `--num_machines` was set to a value of `1`
        `--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
Trying to import sageattention
Successfully imported sageattention
INFO:__main__:Load dataset config from \Downloads\text-to-video-training\musubi-tuner\dataset.toml
INFO:dataset.image_video_dataset:glob images in /Downloads/text-to-video-training/training_data/chj
INFO:dataset.image_video_dataset:found 25 images
INFO:dataset.config_utils:[Dataset 0]
  is_image_dataset: True
  resolution: (960, 544)
  batch_size: 1
  num_repeats: 4
  caption_extension: ".txt"
  enable_bucket: True
  bucket_no_upscale: False
  cache_directory: "/Downloads/text-to-video-training/training_data/cha"
  debug_dataset: False
    image_directory: "/Downloads/text-to-video-training/training_data/chj"
    image_jsonl_file: "None"


INFO:dataset.image_video_dataset:bucket: (560, 928), count: 4
INFO:dataset.image_video_dataset:bucket: (592, 880), count: 12
INFO:dataset.image_video_dataset:bucket: (608, 848), count: 8
INFO:dataset.image_video_dataset:bucket: (640, 816), count: 44
INFO:dataset.image_video_dataset:bucket: (704, 736), count: 4
INFO:dataset.image_video_dataset:bucket: (720, 720), count: 20
INFO:dataset.image_video_dataset:bucket: (768, 672), count: 4
INFO:dataset.image_video_dataset:bucket: (880, 592), count: 4
INFO:dataset.image_video_dataset:total batches: 100
INFO:__main__:preparing accelerator
accelerator device: cuda
INFO:__main__:DiT precision: torch.bfloat16, weight precision: torch.float8_e4m3fn
INFO:__main__:Loading DiT model from \Downloads\text-to-image-video\ComfyUI\models\diffusion_models\mp_rank_00_model_states.pt
Using torch attention mode, split_attn: False
import network module: networks.lora
INFO:networks.lora:create LoRA network. base dim (rank): 32, alpha: 1
INFO:networks.lora:neuron dropout: p=None, rank dropout: p=None, module dropout: p=None
INFO:networks.lora:create LoRA for U-Net/DiT: 240 modules.
INFO:networks.lora:enable LoRA for U-Net: 240 modules
HYVideoDiffusionTransformer: Gradient checkpointing enabled.
prepare optimizer, data loader etc.
INFO:__main__:use 8-bit AdamW optimizer | {}
override steps. steps for 20 epochs is / 指定エポックまでのステップ数: 2000
INFO:__main__:casting model to torch.float8_e4m3fn
running training / 学習開始
  num train items / 学習画像、動画数: 100
  num batches per epoch / 1epochのバッチ数: 100
  num epochs / epoch数: 20
  batch size per device / バッチサイズ: 1
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 2000
INFO:__main__:set DiT model name for metadata: \Downloads\text-to-image-video\ComfyUI\models\diffusion_models\mp_rank_00_model_states.pt
steps:   0%|                                                                                                                                             | 0/2000 [00:00<?, ?it/s]INFO:__main__:DiT dtype: torch.float8_e4m3fn, device: cuda:0

epoch 1/20
Trying to import sageattention
Successfully imported sageattention
Trying to import sageattention
Successfully imported sageattention
INFO:dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
INFO:dataset.image_video_dataset:epoch is incremented. current_epoch: 0, epoch: 1
steps:   5%|█████▋                                                                                                           | 100/2000 [10:27<3:18:41,  6.27s/it, avr_loss=0.112]
epoch 2/20
INFO:dataset.image_video_dataset:epoch is incremented. current_epoch: 1, epoch: 2
INFO:dataset.image_video_dataset:epoch is incremented. current_epoch: 1, epoch: 2
steps:  10%|███████████▎                                                                                                     | 200/2000 [20:53<3:07:58,  6.27s/it, avr_loss=0.116]

...

epoch 19/20
INFO:dataset.image_video_dataset:epoch is incremented. current_epoch: 18, epoch: 19
INFO:dataset.image_video_dataset:epoch is incremented. current_epoch: 18, epoch: 19
steps:  95%|█████████████████████████████████████████████████████████████████████████████████████████████████████████▍     | 1900/2000 [3:18:49<10:27,  6.28s/it, avr_loss=0.0816]
epoch 20/20
INFO:dataset.image_video_dataset:epoch is incremented. current_epoch: 19, epoch: 20
INFO:dataset.image_video_dataset:epoch is incremented. current_epoch: 19, epoch: 20
steps: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2000/2000 [3:29:22<00:00,  6.28s/it, avr_loss=0.087]
saving checkpoint: \Downloads\text-to-video-training\trained_lora\chj-lora.safetensors
INFO:__main__:model saved.
steps: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2000/2000 [3:29:23<00:00,  6.28s/it, avr_loss=0.087]
```

```
PS \Downloads\text-to-video-training\musubi-tuner> python convert_lora.py --input \Downloads\text-to-video-training\trained_lora\chj-lora.safetensors --output  \Downloads\text-to-image-video\ComfyUI\models\loras\hunyuan_trained\chj.safetensors --target other
INFO:__main__:loading  \Downloads\text-to-video-training\trained_lora\chj-lora.safetensors
INFO:__main__:converting to other
INFO:__main__:saving to \Downloads\text-to-image-video\ComfyUI\models\loras\hunyuan_trained\chj.safetensors
INFO:__main__:done
```

So, I trained with 25 good quality images with captioned .txt files having the same name with the corresponding image file. I tried both comfyUI and  `python hv_generate_video.py ...` and none of them actually affect the video. Can you kindly help me to figure out what was the problem? By the way, thanks a lot to make and share this repo!

p.s.: I also trained a lora with `learning rate of 2e-4`, `--timestep_sampling shift`, and `--discrete_flow_shift 7.0`. and it did not affect the video.

**Comment by taekim99** [comment]
`python hv_generate_video.py --fp8 --video_size 544 960 --video_length 5 --infer_steps 6 --prompt "woman walks in NYC, white luxury dress, video focusing on her, ultra-realistic, 4K resolution, highly detailed textures, cinematic lighting, realistic skin and fabric details."  --save_path \Downloads\text-to-video-training --output_type both --dit \Downloads\text-to-image-video\ComfyUI\models\diffusion_models\mp_rank_00_model_states.pt --attn_mode sdpa --split_attn --vae \Downloads\text-to-image-video\ComfyUI\models\vae\hunyuan_video_vae_bf16.safetensors --vae_chunk_size 32 --vae_spatial_tile_sample_min_size 128 --text_encoder1 \Downloads\text-to-image-video\ComfyUI\models\text_encoders\llava_llama3_fp16.safetensors --text_encoder2 \Downloads\text-to-image-video\ComfyUI\models\text_encoders\clip_l_hunyuan.safetensors --seed 1234 --lora_multiplier 1.0 --lora_weight \taehy\Downloads\text-to-video-training\trained_lora\chj-lora.safetensors`

pretty much same with the one in readme. Also, I found that it generates a 0 sec video? does `--video_length 5` make 5 sec video or 5 frames?

**Comment by taekim99** [comment]
simply re-trained a lora and now it works well? idk why though. but problem solved.
